numpy/lib/format.py

```diff
@@ -590,7 +590,8 @@ def _read_array_header(fp, version):
     #   "shape" : tuple of int
     #   "fortran_order" : bool
     #   "descr" : dtype.descr
-    header = _filter_header(header)
+    if version <= (2, 0):
+        header = _filter_header(header)
```

Comment:
OK, I doubt it is very useful, but its simple enough.  If you are enthusiastic, maybe add a comment why we only do this on older "versions", though?
(Would be happy to just skip as well).

Reply:
It's useful because it lets me pair a `write_array(..., version=(3,0)` with `read_array(..., version=(3,0))` and get the nice perf boost 😎

Reply:
Hah, fair enough :). Thanks @ohadravid!